### DISCUSSION TITLE
parallel access to one device on windows

### DISCUSSION THREAD

**Discussion by denbin1** [christoph2/pyxcp#162] [General]
Hallo everyone,

I use python "canopen" to simulate can devices for a device which is under test (DUT). The DUT also has a XCP interface and I'm also able to communicate with the DUT using "pyxcp".

What I would like now is to combine both functions - the canopen simulation and the xcp interface - into one "python script". I think this is not possible at the moment, bacause both can protocols (the canopen and the xcp) do open and connect to the can bus.
So the later gets a busy device.

How can I integrate XCP into my working canopen simulation ?
I'm working with windows and I think this would not be a problem when working on Linux.

I would be very glad if someone has an idea.

Regards,
Dennis

**Comment by christoph2** [comment]
Have you tried it and got an error?
If so, Python 3.13 multiple interpreters (module `interpreters`) may be a solution (not only) to this problem;
it think this feature will also enable multiple pyXCP connections at the same time.

**Comment by denbin1** [comment]
Hello Christoph,

thanks for your reply. Yes I tried this. But initialization of xcp fails (on windows).
I thought of something like: 
Maybe its possible to initialize the xcp with a bus handle which has is already been initialized.

regards,
Dennis

**Comment by christoph2** [comment]
It could be made possible, yes!
I'm currently working on a massive update, which includes a new traitlets based configuration system, i.e. the configuration file is basically Python. 
For example, one could use a seed-and-key Python function, if the algorithm is known (tested /w CANape).
I'll let you know, if I find a appropriate solution -- good ideas are always welcome...
